app/forms/hyrax/forms/resource_batch_edit_form.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+module Hyrax
+  module Forms
+    class ResourceBatchEditForm < Hyrax::Forms::ResourceForm
+      include Hyrax::FormFields(:basic_metadata)
+
+      self.required_fields = []
+      self.model_class = Hyrax.primary_work_type
+
+      # Contains a list of titles of all the works in the batch
+      attr_accessor :names
+
+      # @param [Hyrax::Work] model the model backing the form
+      # @param [Ability] current_ability the user authorization model
+      # @param [Array<String>] batch_document_ids a list of document ids in the batch
+      def initialize(model, _current_ability, batch_document_ids)
+        @names = []
+        @batch_document_ids = batch_document_ids
+        if @batch_document_ids.present?
+          super(model.class.new(initialize_combined_fields))
+        else
+          super(model)
+        end
+      end
+
+      def terms
+        [:creator, :contributor, :description,
+         :keyword, :resource_type, :license, :publisher, :date_created,
+         :subject, :language, :identifier, :based_near,
+         :related_url]
+      end
+
+      attr_reader :batch_document_ids
+
+      # Returns a list of parameters we accept from the form
+      # rubocop:disable Metrics/MethodLength
+      def self.build_permitted_params
```

Comment:
Question: I just don't know much about hyrax forms anymore - if there's form objects with a restricted subset of fields, why do they also need to define their permitted params? Where's this get used - do all the form objects have these or is it something special about bulk edit?

Reply:
@tpendragon you probably know more about this form setup than most of us. these are valkyrie `ChangeSet`s.

i have the same question about why this is needed, and where it's used other than the tests below.